coderd/database/migrations/000335_ai_tasks.up.sql

```diff
@@ -0,0 +1,101 @@
+-- Determines if a coder_ai_task resource was included in a
+-- workspace build.
+ALTER TABLE
+    workspace_builds
+ADD
+    COLUMN has_ai_task BOOLEAN NOT NULL DEFAULT FALSE;
+
+-- The app that is displayed in the ai tasks sidebar.
+ALTER TABLE
+    workspace_builds
+ADD
+    COLUMN ai_tasks_sidebar_app_id UUID DEFAULT NULL;
+
+ALTER TABLE
+    workspace_builds
+ADD
+    CONSTRAINT workspace_builds_ai_tasks_sidebar_app_id_fkey FOREIGN KEY (ai_tasks_sidebar_app_id) REFERENCES workspace_apps(id);
+
+-- Determines if a coder_ai_task resource is defined in a template version.
+ALTER TABLE
+    template_versions
+ADD
+    COLUMN has_ai_task BOOLEAN NOT NULL DEFAULT FALSE;
+
+-- The Tasks tab will be rendered in the UI only if there's at least one template version with has_ai_task set to true.
+-- The query to determine this will be run on every UI render, and this index speeds it up.
+-- SELECT EXISTS (SELECT 1 FROM template_versions WHERE has_ai_task = TRUE);
+CREATE INDEX idx_template_versions_has_ai_task ON template_versions USING btree (has_ai_task);
+
+DROP VIEW workspace_build_with_user;
+
+CREATE VIEW workspace_build_with_user AS
```

Comment:
It'd be helpful if you pointed out to the future reader (in a comment) what you've modified in the view.